agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/StartupDiagnostics.java

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.applicationinsights.agent.internal.init;
+
+import com.microsoft.applicationinsights.agent.bootstrap.diagnostics.PidFinder;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+
+class StartupDiagnostics {
+
+  // For Java >= 9
+  // Execute with -XX:+UnlockDiagnosticVMOptions -XX:CompilerDirectivesFile=path/jvm-disable-c2.json
+  // Content of jvm-disable-c2.json file:
+  // [
+  //  {
+  //    match: [
+  //      "*::*"
+  //    ],
+  //    c2: {
+  //      Exclude: true
+  //    }
+  //  }
+  // ]
+  private static final String APPPLICATIONINSIGHTS_DEBUG_JIT_C_2_DISABLING_ENABLED =
+      "appplicationinsights.debug.jit.c2-disabling.enabled";
+
+  private static final String APPPLICATIONINSIGHTS_DEBUG_JVM_FLAGS_ENABLED =
+      "appplicationinsights.debug.jvm-flags.enabled";
+
+  // Execute with -XX:NativeMemoryTracking=summary
+  private static final String APPPLICATIONINSIGHTS_DEBUG_NATIVE_MEM_TRACKING_ENABLED =
+      "appplicationinsights.debug.native-mem-tracking.enabled";
+
+  // "file" (default value) / "console" / "file-console"
+  public static final String APPPLICATIONINSIGHTS_DEBUG_DIAG_EXPORT =
+      "appplicationinsights.debug.diag-export";
```

Comment:
see comment below, maybe we don't need any of these, and can enable them all automatically at self-diagnostic `debug` level?

Reply:
@trask I reply to your two comments about self-diagnostic and the properties added in this PR.

The native memory tracking requires the addition of the `-XX:NativeMemoryTracking=summary` JVM option. I would prefer that the current self diagnostics features remains as simple as possible to configure for our users.

I would like to be able to execute only one type of memory diag (rss, native mem tracking, and perhaps one day heap allocation) to avoid that a measure comes to impact the result of another one. I would also like to avoid that an excessive logging from the self-diagnostic feature comes to pollute the memory measurements. 

We can have different levels of diagnostics. First, the self diagnostics, simple to use. At a second level, the start up profiler and these memory diagnostics at start-up.

For all these reasons, I would prefer to separate the features of this PR from the current self diagnostic.

I can propose to create another PR to add the rss measurement to the current self-diagnostic, to have an idea of the value. _Could you please give me a pointer where to put the code?_ 

> similar to how we dump all jmx metrics at that level

 I did not know that. _Could you please give me a pointer in the code?_ I have removed the retrieval of the JVM flags from this PR. If the current jmx metrics do not contain them, I will add them in another PR. I will also add the classpath content if it is not already retrieved.